src/modules/bpmn-diff-view/bpmn-diff-view.ts

```
@@ -0,0 +1,371 @@
+  */
+  private _removeElementsWithoutChanges(changedElements: object): object {
+    for (const elementId in changedElements) {
+      const currentElementHasNoChanges: boolean = Object.keys(changedElements[elementId].attrs).length === 0;
+
+      if (currentElementHasNoChanges) {
+        delete changedElements[elementId];
+      }
+    }
+
+    return changedElements;
+  }
+
+  private _updateDiffView(): void {
+    if (this.currentDiffMode === DiffMode.AfterVsBefore) {
+      this._updateLowerDiff(this.xml);
+      this.diffModeTitle = 'After vs. Before';
+    } else if (this.currentDiffMode === DiffMode.BeforeVsAfter) {
+      this._updateLowerDiff(this.savedxml);
+      this.diffModeTitle = 'Before vs. After';
+    } else {
+      this.diffModeTitle = '';
+    }
+  }
+
+  private async _updateLowerDiff(xml: string): Promise<void> {
+    const xmlIsNotLoaded: boolean = xml === undefined || xml === null;
+
+    if (xmlIsNotLoaded) {
+      return;
+    }
+
+    const addedElements: Object = this.changes._added;
+    const removedElements: object = this.changes._removed;
+    const changedElements: object = this.changes._changed;
+    const layoutChangedElements: object = this.changes._layoutChanged;
+
+    const diffModeIsAfterVsBefore: boolean = this.currentDiffMode === DiffMode.AfterVsBefore;
+
+    await this._importXml(xml, this._diffModeler);
+    this._markLayoutChangedElements(layoutChangedElements);
+    this._markChangedElements(changedElements);
+
+    diffModeIsAfterVsBefore ?
+      this._markAddedElements(addedElements) :
+      this._markRemovedElements(removedElements);
+
+    const coloredXml: string = await this._getXmlFromModeler();
+    await this._importXml(coloredXml, this._lowerViewer);
+  }
+
+  private _importXml(xml: string, viewer: IBpmnModeler): Promise <void> {
+    const xmlIsNotLoaded: boolean = xml === undefined || xml === null;
+
+    if (xmlIsNotLoaded) {
+      return;
```

Is this case a _normal_ case or can we throw an error here?